packages/next/client/link.tsx

```diff
@@ -18,6 +18,7 @@ import {
 import { useIntersection } from './use-intersection'
 import { getDomainLocale } from './get-domain-locale'
 import { addBasePath } from './add-base-path'
+import { getSegmentParam } from '../server/utils'
```

Comment:
Not a fan of this because this is not a generic util to parse the dynamic parameter in a client component, it's intentionally not optimized for bundle size. Can the check be simpler instead. e.g. if holding brackets in `href` error in `app`.

Reply:
Updated. Currently only check `href.pathname`, should `href="/[slug]"` error as well? 🤔